File: vivuvn_api/vivuvn_api/Models/ItineraryDay.cs
using System.ComponentModel.DataAnnotations;

namespace vivuvn_api.Models
{
    public class ItineraryDay
    {
        [Key]
        public int Id { get; set; }

        public int ItineraryId { get; set; }
        public Itinerary Itinerary { get; set; } = null!;

        public int DayNumber { get; set; }
        public DateTime Date { get; set; }

        public ICollection<ItineraryItem> Items { get; set; } = new List<ItineraryItem>();
        public ICollection<ExternalService> ExternalServices { get; set; } = new List<ExternalService>();
    }
}

File: vivuvn_app/lib/features/itinerary/itinerary-detail/schedule/model/itinerary_day.dart
import 'itinerary_item.dart';

class ItineraryDay {
  final int id;
  final int dayNumber;
  final DateTime? date;
  final List<ItineraryItem> items;

  ItineraryDay({
    required this.id,
    required this.dayNumber,
    required this.date,
    required this.items,
  });

  factory ItineraryDay.fromJson(final Map<String, dynamic> json) {
    return ItineraryDay(
      id: json['id'] is int ? json['id'] as int : 0,
      dayNumber: json['dayNumber'] is int ? json['dayNumber'] as int : 0,
      date: json['date'] != null ? DateTime.tryParse(json['date']) : null,
      items:
          (json['items'] as List?)
              ?.map((final e) => ItineraryItem.fromJson(e))
              .toList() ??
          [],
    );
  }

  // ==== Thêm copyWith ====
  ItineraryDay copyWith({
    final int? id,
    final int? dayNumber,
    final DateTime? date,
    final List<ItineraryItem>? items,
  }) {
    return ItineraryDay(
      id: id ?? this.id,
      dayNumber: dayNumber ?? this.dayNumber,
      date: date ?? this.date,
      items: items ?? this.items,
    );
  }
}
